src/lib.rs

```diff
@@ -74,19 +76,36 @@ struct SandboxFS {
 
 impl SandboxFS {
     /// Creates a new `SandboxFS` instance.
-    fn new() -> SandboxFS {
+    fn new(mappings: &Vec<Mapping>) -> io::Result<SandboxFS> {
         let root = {
-            let now = time::get_time();
-            let uid = unsafe { libc::getuid() } as u32;
-            let gid = unsafe { libc::getgid() } as u32;
-            nodes::Dir::new_root(now, uid, gid)
+            if mappings.len() == 0 {
+                let now = time::get_time();
+                let uid = unsafe { libc::getuid() } as u32;
```

Comment:
Consider bringing in a crate like `nix` to avoid the unsafe use here.

Reply:
I gave this a try... but `nix`'s `getpid` returns a `Uid` and I don't see a way to cast that back to a raw integer number, which I need later. Am I missing something obvious?

Also, while `nix` contains some useful functions that would avoid me having to make my own (like `mkfifo` in the `conv` tests), that's about it so far. I did not see replacements for all other functions I need to call via libc. In other words: I don't think using `nix` *right now* is a good idea because I cannot eliminate all references to `libc`; prefer to keep a single pattern.

Reply:
I've filed https://github.com/nix-rust/nix/issues/943 about the UID/GID issue. I've also filed a PR to implement `chmod` and was told there is a way to do this natively within Rust. I'll continue investigating what will it take to avoid calling directly into libc for this project. But for now I'd rather stick to libc until I can supply a PR that kills all usages in favor of nix.